presto-main/src/test/java/io/prestosql/sql/analyzer/TestAnalyzer.java

```diff
@@ -253,10 +256,42 @@ public void testHavingReferencesOutputAlias()
     }
 
     @Test
-    public void testWildcardWithInvalidPrefix()
+    public void testSelectAllColumns()
     {
+        // wildcard without FROM
+        assertFails("SELECT *")
+                .hasErrorCode(COLUMN_NOT_FOUND);
+
+        // wildcard with invalid prefix
         assertFails("SELECT foo.* FROM t1")
                 .hasErrorCode(TABLE_NOT_FOUND);
+
+        assertFails("SELECT a.b.c.d.* FROM t1")
+                .hasErrorCode(TABLE_NOT_FOUND);
+
+        // name aliases are not allowed when selecting all columns from a table
```

Comment:
This should say "not currently supported"? It should be allowed from the `If N ≤ 3 and PIC_N is equivalent to an exposed <table or query name> that is in scope` case. It's actually one of the best uses for this feature. If you want to rename all the columns from a table you currently have to do `SELECT c1 AS n1, c2 as n2, c3 as n3, ...`. With this, it can be simplified for `SELECT t.* AS (n1, n2, n3)` and save some typing.

Reply:
Apparently I got trapped (again) following the preexisting behavior.
Adding support for aliases when `IC` refers to a table.